From 321a176ad7a80402c244e8194c999ae8de126f07 Mon Sep 17 00:00:00 2001
From: Brian Lowe <mc-hobbit@groupbcl.ca>
Date: Sat, 14 Jan 2017 20:18:03 -0600
Subject: [PATCH] Tickets #3681 and #3742

Ticket #3681: Document when changes to mc.ext take effect.
Ticket #3742: Update comment block at the start of the file.

Signed-off-by: Brian Lowe <mc-hobbit@groupbcl.ca>
---
 misc/mc.ext.in | 207 +++++++++++++++++++++++++++++++++------------------------
 1 file changed, 119 insertions(+), 88 deletions(-)

diff --git a/misc/mc.ext.in b/misc/mc.ext.in
index 92f908b..a14a984 100644
--- a/misc/mc.ext.in
+++ b/misc/mc.ext.in
@@ -1,107 +1,138 @@
-# Midnight Commander 3.0 extension file
-# Warning: Structure of this file has changed completely with version 3.0
+# Midnight Commander extension file
+#
+# Empty lines and lines starting with "#" are ignored.
+#
+# IMPORTANT: mc scans this file only at startup or after updating it using the
+# mc "Edit extension file" command (F9-c-e). If you edit the file otherwise
+# while mc is running, you will need to (F9-c-e) and exit the editor for your
+# changes to take effect, or exit mc and restart.
 #
-# All lines starting with # or empty lines are thrown away.
 # Lines starting in the first column should have following format:
 #
-# keyword/descNL, i.e. everything after keyword/ until new line is desc
+#   KEYWORD/DESC<NL> (i.e. everything after "KEYWORD/" until newline is "DESC")
 #
-# keyword can be:
+# KEYWORD is one of the following:
 #
-#    shell (desc is, when starting with a dot, any extension (no wildcars),
-#          i.e. matches all the files *desc . Example: .tar matches *.tar;
-#          if it doesn't start with a dot, it matches only a file of that name)
+#    shell
+#       DESC is, when starting with a dot, an extension (no wildcards.)
+#       All the files matching '*DESC'. Example: .tar matches *.tar; if it
+#       doesn't start with a dot, it matches only a file of that name.
 #
-#    shell/i (desc is, when starting with a dot, any extension (no wildcars),
-#          The same as shell but with case insensitive.
+#    shell/i
+#       DESC is, when starting with a dot, an extension (no wildcards.)
+#       The same as "shell" but case-insensitive.
 #
-#    regex (desc is an extended regular expression)
-#          Please note that we are using the GNU regex library and thus
-#          \| matches the literal | and | has special meaning (or) and
-#          () have special meaning and \( \) stand for literal ( ).
+#    regex
+#       DESC is an extended regular expression.
+#       Note that mc uses the GNU regex library and thus:
+#         | has a special meaning ("or") and \| matches the literal |
+#         () have special meaning and \( \) stand for literal ( )
 #
-#    regex/i (desc is an extended regular expression)
-#          The same as regex but with case insensitive.
+#    regex/i
+#       DESC is an extended regular expression.
+#       The same as "regex" but case-insensitive.
 #
-#    type  (file matches this if `file %f` matches regular expression desc
-#          (the filename: part from `file %f` is removed))
+#    type
+#       File matches this if `file %f` matches regular expression DESC
+#       (the 'filename:' text from `file %f` is removed)
 #
-#    type/i (file matches this if `file %f` matches regular expression desc)
-#          The same as type but with case insensitive.
+#    type/i
+#       File matches this if `file %f` matches regular expression DESC
+#       The same as "type" but case-insensitive.
 #
-#    directory (matches any directory matching regular expression desc)
+#    directory
+#       Matches any directory matching regular expression DESC
 #
-#    include (matches an include directive)
+#    include
+#       Matches an 'include' directive
 #
-#    default (matches any file no matter what desc is)
+#    default
+#       Matches any file regardless of the value of DESC
 #
 # Other lines should start with a space or tab and should be in the format:
 #
-# keyword=commandNL (with no spaces around =), where keyword should be:
-#
-#    Open (if the user presses Enter or doubleclicks it),
-#
-#    View (F3), Edit (F4)
-#
-#    Include is the keyword used to add any further entries from an include/
-#    section
-#
-# command is any one-line shell command, with the following substitutions:
-#
-# %% -> % character
-# %p -> name of the current file (without path, but pwd is its path).
-#       Also provided to external application as MC_EXT_BASENAME
-#       global variable
-# %f -> name of the current file. Unlike %p, if file is located on a
-#	non-local virtual filesystem, i.e. either tarfs or ftpfs,
-#	then the file will be temporarily copied into a local directory
-#	and %f will be the full path to this local temporal file.
-#	If you don't want to get a local copy and want to get the
-#	virtual fs path (like /#ftp:ftp.cvut.cz/pub/hungry/xword), then
-#	use %d/%p instead of %f.
-#       Also provided to external application as MC_EXT_FILENAME
-#       global variable
-# %d -> name of the current directory (pwd, without trailing slash)
-#       Also provided to external application as MC_EXT_CURRENTDIR
-#       global variable
-# %s -> "selected files", i.e. space separated list of tagged files if any
-#       or name of the current file.
-#       Also provided to external application as MC_EXT_SELECTED
-#       global variable
-# %t -> list of tagged files
-#       Also provided to external application as MC_EXT_ONLYTAGGED
-#       global variable
-# %u -> list of tagged files (they'll be untaged after the command)
-#
-# (If these 6 letters are in uppercase, they refer to the other panel.
-# But you shouldn't have to use it in this file.)
-#
-#
-# %cd -> the rest is a path mc should change into (cd won't work, since it's
-#	a child process).  %cd handles even vfs names.
-#
-# %view -> the command you type will be piped into mc's internal file viewer
-#	if you type only the %view and no command, viewer will load %f file
-#	instead (i.e. no piping, so it is different to %view cat %f)
-#	%view may be directly followed by {} with a list of any of
-#	ascii (Ascii mode), hex (Hex mode), nroff (color highlighting for
-#	text using backspace for bold and underscore) and unform
-#	(no highlighting for nroff sequences) separated by commas.
-#
-# %var -> You use it like this: %var{VAR:default}.  This macro will expand
-#       to the value of the VAR variable in the environment if it's set
-#       otherwise the value in default will be used.  This is similar to
-#       the Bourne shell ${VAR-default} construct.
-#
-# Rules are applied from top to bottom, thus the order is important.
-# If some actions are missing, search continues as if this target didn't
-# match (i.e. if a file matches the first and second entry and View action
-# is missing in the first one, then on pressing F3 the View action from
-# the second entry will be used. default should catch all the actions.
-#
-# Any new entries you develop for you are always welcome if they are
-# useful on more than one system.  You can post your modifications
-# as tickets at www.midnight-commander.org
+#       KEYWORD=COMMAND<NL> (with no spaces around =)
+#
+# KEYWORD is one of the # following:
+#
+#    Open
+#       COMMAND is run if the user presses Enter or double-clicks the file
+#
+#    View
+#       COMMAND is run if the user presses F3
+#
+#    Edit
+#       COMMAND is run if the user presses F4
+#
+#    Include
+#       Add any further entries from an 'include/' section
+#
+# COMMAND is any one-line shell command, with the following substitutions:
+#
+#    %%
+#       The '%' character
+#
+#    %p or $MC_EXT_BASENAME
+#       Name of the current file (without path, but 'pwd' is its path.)
+#
+#    %f or $MC_EXT_FILENAME
+#       Name of the current file. Unlike %p, if file is located on a non-local
+#       virtual filesystem (e.g. tarfs or ftpfs,) then the file will be
+#       temporarily copied into a local directory and %f will be the full path
+#       to this local temporal file.  If you don't want to get a local copy and
+#       want to get the virtual fs path (like ftp://ftp.cvut.cz/pub/linux), 
+#       then use %d/%p instead of %f.
+#
+#    %d or $MC_EXT_CURRENTDIR
+#       Name of the current directory (pwd, without trailing slash)
+#
+#    %s or $MC_EXT_SELECTED
+#       Space separated list of tagged files (if any) or the name of the
+#       current file.
+#
+#    %t or $MC_EXT_ONLYTAGGED
+#       Space-spearted list of tagged files
+#
+#    %u
+#       list of tagged files; untagged after command completes
+#
+#    (If the letter following the % in the above six lines is in uppercase, it
+#    refers to the other panel.  But you shouldn't have to use it in this
+#    file.)
+#
+#    %cd PATH
+#       PATH that mc should change into ("cd" won't work because it's a child
+#       process.)  %cd handles even vfs names.
+#
+#    %view
+#       Output from COMMAND will be piped into mc's internal file viewer.  If
+#       you type only '%view' and no COMMAND, viewer will directly load the %f
+#       file without piping.  The result is the same as if COMMAND was "%view
+#       cat %f," but without the need to fork a shell, run 'cat', and read its
+#       output.
+#
+#       %view may be directly followed by {} with one or more of the following
+#       separated by commas:
+#         ascii  (ascii mode)
+#         hex    (hex mode)
+#         nroff  (colour highlighting for text using char<backspace>char
+#                for bold and <underscore><backspace>char for underscore)
+#         unform (no highlighting for nroff sequences)
+#
+#    %var{VARNAME:default}
+#       Expands to the value of the VARNAME variable in the environment if it's
+#       set, otherwise the value in "default" is used. This is similar to the
+#       Bourne shell ${VAR-default} construct.
+#
+# Rules are applied from top to bottom, thus the order is important.  If some
+# actions are missing, search continues as if this target didn't match (i.e. if
+# a file matches the first and second entry and View action is missing in the
+# first one, then on pressing F3 the View action from the second entry will be
+# used.)
+#
+# Any new entries you develop for you are always welcome if they are useful
+# on more than one system.  You can post your modifications as tickets at
+# www.midnight-commander.org
 
 
 ### Changes ###
-- 
2.5.5

